client/pages/roles.tsx

```diff
@@ -1,61 +1,63 @@
-import { Card, Divider, Grid, Page, Spacer, Text } from '@geist-ui/core';
+
+import { Card, Divider, Grid, Link, Page, Spacer, Text } from '@geist-ui/core';
 import type { NextPage } from 'next';
 import { useEffect, useState } from 'react';
-import { Role } from '../database/roles';
+import { Role } from './api/roles';
 
 interface RoleAPI {
     Role_ID: string;
     Role_Name: string;
     Role_Description: string;
 }
 
+
 const ViewRolePage: NextPage = () => {
     const [roles, setRoles] = useState<Role[]>();
 
     const Role = ({
         id,
-        label,
+        name,
         description,
     }: {
         id: string;
-        label: string;
+        name: string;
         description: string;
     }) => {
         return (
             <Grid justify="center">
-                <Card width="600px" key={label}>
+                <Card width="600px" key={name}>
                     <Card.Content>
                         <Text b my={0}>
-                            {label}
+                            {name}
                         </Text>
                     </Card.Content>
                     <Divider h="1px" my={0} />
                     <Card.Content>
-                        <Text>
-                            {description} (Role ID: {id})
+                        <Text font="12px" mt={0} style={{ color: '#ccc' }}>
+                            DESCRIPTION
                         </Text>
+                        <Text>{description}</Text>
                     </Card.Content>
+                    <Card.Footer>
+                        <Link color href={'enroll/' + id}>
+                            View skills for {name} &#8594;
+                        </Link>
+                    </Card.Footer>
                 </Card>
                 <Spacer h={2} />
             </Grid>
         );
     };
 
     useEffect(() => {
-        fetch(process.env.API_URL + '/roles')
+        fetch('/api/roles', { method: 'GET' })
             .then((response) => response.json())
             .then((result) => {
-                const rolesResult = result.map((row: RoleAPI) => {
-                    return {
-                        id: Number(row.Role_ID),
-                        label: row.Role_Name,
-                        description: row.Role_Description,
-                    };
-                });
-                setRoles(rolesResult);
+                console.log({ result });
+                setRoles(result);
             })
             .catch((error) => console.log('error', error));
-    });
+    }, []);
```

Comment:
The API call logic is placed within a `useEffect()` hook here.

Change the `fetch(ENDPOINT, {method: 'GET/POST/PUT/DELETE'})` parameters accordingly.